scripts/exregional_process_lightning.sh

```diff
@@ -0,0 +1,317 @@
+# TRUE.
+#
+#-----------------------------------------------------------------------
+#
+print_input_args valid_args
+#
+#-----------------------------------------------------------------------
+#
+# Load modules.
+#
+#-----------------------------------------------------------------------
+#
+case $MACHINE in
+#
+"WCOSS_C" | "WCOSS")
+#
+
+  if [ "${USE_CCPP}" = "TRUE" ]; then
+  
+# Needed to change to the experiment directory because the module files
+# for the CCPP-enabled version of FV3 have been copied to there.
+
+    cd_vrfy ${CYCLE_DIR}
+  
+    set +x
+    source ./module-setup.sh
+    module use $( pwd -P )
+    module load modules.fv3
+    module list
+    set -x
+  
+  else
+  
+    . /apps/lmod/lmod/init/sh
+    module purge
+    module use /scratch4/NCEPDEV/nems/noscrub/emc.nemspara/soft/modulefiles
+    module load intel/16.1.150 impi/5.1.1.109 netcdf/4.3.0 
+    module list
+  
+  fi
+
+  ulimit -s unlimited
+  ulimit -a
+  APRUN="mpirun -l -np ${PE_MEMBER01}"
+  ;;
+#
+"HERA")
+  ulimit -s unlimited
+  ulimit -a
+  APRUN="srun"
+  LD_LIBRARY_PATH="${UFS_WTHR_MDL_DIR}/FV3/ccpp/lib${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}"
+  ;;
+#
+"JET")
+  ulimit -s unlimited
+  ulimit -a
+  APRUN="srun"
+  LD_LIBRARY_PATH="${UFS_WTHR_MDL_DIR}/FV3/ccpp/lib${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}"
+  ;;
+#
+"ODIN")
+#
+  module list
+
+  ulimit -s unlimited
+  ulimit -a
+  APRUN="srun -n ${PE_MEMBER01}"
+  ;;
+#
+esac
+#
+#-----------------------------------------------------------------------
+#
+# Extract from CDATE the starting year, month, day, and hour of the
+# forecast.  These are needed below for various operations.
+#
+#-----------------------------------------------------------------------
+#
+set -x
+START_DATE=`echo "${CDATE}" | sed 's/\([[:digit:]]\{2\}\)$/ \1/'`
+  YYYYMMDDHH=`date +%Y%m%d%H -d "${START_DATE}"`
+  JJJ=`date +%j -d "${START_DATE}"`
+
+YYYY=${YYYYMMDDHH:0:4}
+MM=${YYYYMMDDHH:4:2}
+DD=${YYYYMMDDHH:6:2}
+HH=${YYYYMMDDHH:8:2}
+YYYYMMDD=${YYYYMMDDHH:0:8}
+
+YYJJJHH=`date +"%y%j%H" -d "${START_DATE}"`
+PREYYJJJHH=`date +"%y%j%H" -d "${START_DATE} 1 hours ago"`
+
+#
+#-----------------------------------------------------------------------
+#
+# Get into working directory
+#
+#-----------------------------------------------------------------------
+#
+print_info_msg "$VERBOSE" "
+Getting into working directory for lightning process ..."
+
+cd ${WORKDIR}
+
+fixdir=$FIXgsi/
+
+print_info_msg "$VERBOSE" "fixdir is $fixdir"
+
+#
+#-----------------------------------------------------------------------
+#
+# link or copy background files
+#
+#-----------------------------------------------------------------------
+
+cp_vrfy ${fixdir}/fv3_grid_spec          fv3sar_grid_spec.nc
+cp_vrfy ${fixdir}/geo_em.d01.nc          geo_em.d01.nc
+
+
+#
+# Link to the NLDN data
+#
+#-----------------------------------------------------------------------
+filenum=0
+LIGHTNING_FILE=${LIGHTNING_ROOT}/vaisala/netcdf
```

Comment:
There is a lot of duplication of information in this section, leaving a lot of room for error if any of these file names change in the future. Please consider making this a loop, for example:

```
for n in 00 05 35 40 45 50 55 ; do
  filename=${LIGHTNING_FILE}/${YYJJJHH}${n}0005r
  if [ -r ${filename ]; then
  ((filenum += 1 ))
    ln -sf ${filename} ./NLDN_lightning_${filenum}
  else
   echo " ${filename} does not exist"
  fi
done
```

Reply:
Done. Looks much better.